app/helpers/sort_helper.rb

```diff
@@ -389,35 +392,43 @@ def find_filter_for_column(column, filter_mapping)
                   allowed_params: nil, **html_options)
-    caption ||= column.to_s.humanize
+    attribute = column.attribute
+    caption ||= attribute.to_s.humanize
 
-    filter = find_filter_for_column(column, filter_column_mapping)
+    filter = find_filter_for_column(attribute, filter_column_mapping)
     sortable = html_options.delete(:sortable)
 
     # `param` is not needed in the `content_arguments`, but should remain in the `html_options`.
     # It is important for keeping the current state in the GET parameters of each link used in
     # the action menu.
     content_args = html_options.merge(rel: :nofollow, param: nil)
 
-    render Primer::Alpha::ActionMenu.new(menu_id: "menu-#{column}") do |menu|
-      action_button(menu, caption, favorite: column == :favored)
+    render Primer::Alpha::ActionMenu.new(menu_id: "menu-#{attribute}") do |menu|
+      action_button(menu, column, caption, favorite: column == :favored)
 
       # Some columns are not sortable or do not offer a suitable filter. Omit those actions for them.
-      sort_actions(menu, column, default_order, content_args:, allowed_params:, **html_options) if sortable
-      filter_action(menu, column, filter, content_args:) if filter
+      sort_actions(menu, attribute, default_order, content_args:, allowed_params:, **html_options) if sortable
+      filter_action(menu, attribute, filter, content_args:) if filter
 
-      move_column_actions(menu, column, table_columns, content_args:, allowed_params:, **html_options)
-      add_and_remove_column_actions(menu, column, table_columns, content_args:, allowed_params:, **html_options)
+      move_column_actions(menu, attribute, table_columns, content_args:, allowed_params:, **html_options)
+      add_and_remove_column_actions(menu, attribute, table_columns, content_args:, allowed_params:, **html_options)
     end
   end
 
-  def action_button(menu, caption, favorite: false)
+  def action_button(menu, column, caption, favorite: false)
+    additional_menu_classes = ["generic-table--action-menu-button",
+                               column.respond_to?(:action_menu_classes) ? column.action_menu_classes : nil]
+                                .compact
+                                .join(" ")
+
     menu.with_show_button(scheme: :link, color: :default, text_transform: :uppercase,
                           underline: false, display: :inline_flex,
-                          classes: "generic-table--action-menu-button") do |button|
+                          classes: additional_menu_classes) do |button|
       if favorite
         # This column only shows an icon, no text.
         render Primer::Beta::Octicon.new(icon: "star-fill", color: :subtle, "aria-label": I18n.t(:label_favorite))
+      elsif column.respond_to?(:action_menu_header)
+        column.action_menu_header(button)
       else
         button.with_trailing_action_icon(icon: :"triangle-down")
 
```

Comment:
The last 2 branches of the condition have a lot in common. I think with a little modification on the `column.action_menu_header` one branch can be removed.  As a bonus, we will have the `h(caption).to_s` correctly html escaped.

```suggestion
      else
        button.with_leading_visual_icon(**column.visual_icon) if column.respond_to?(:visual_icon)
        button.with_trailing_action_icon(icon: :"triangle-down")
```
I also added the [accompanying suggestion](https://github.com/opf/openproject/pull/17400#discussion_r1882738982) to the `def action_menu_header(button)`.